src/metabase/api/activity.clj

```diff
@@ -209,15 +198,28 @@
                                         (mi/can-read? model-object)
                                         ;; hidden tables, archived cards/dashboards
                                         (not (or (:archived model-object)
-                                                 (= (:visibility_type model-object) :hidden))))]
+                                                 (= (:visibility_type model-object) :hidden))))
+                             :let [is-dataset? (= (keyword (:type model-object)) :model)]]
                          (cond-> (assoc view-log :model_object model-object)
-                           (= (keyword (:type model-object)) :model) (assoc :model "dataset")))
+                           is-dataset? (assoc :model "dataset")))
         scored-views (score-items filtered-views)]
     (->> scored-views
-         (sort-by :score)
-         reverse
-         order-items
+         (sort-by
+          ;; sort by model first, and then score when they are the same model
+          (juxt #(-> % :model model->precedence) #(- (% :score))))
          (take 5)
-         (map #(dissoc % :score)))))
+         (map #(-> %
+                   (assoc :timestamp (:max_ts % ""))
+                   recent-views/fill-recent-view-info)))))
```

Comment:
nit - might be a bit more readable to break this out into two separate maps, instead of nested threading macros? and/or to use a transducer for the whole pipeline

Reply:
I landed here because I found the transducer style to be harder to read, though it does have the memory pressure benefit. But then in the other direction: for doing 2 maps we'd be creating another intermediate sequence.